src/mkdocstrings/extension.py

```diff
@@ -231,17 +231,26 @@ def _process_block(
 
 class _PostProcessor(Treeprocessor):
     def run(self, root: Element) -> None:
+        self.remove_duplicated_headings_from_parent(root)
+
+    def remove_duplicated_headings_from_parent(self, parent: Element) -> bool:
```

Comment:
Nitpick: I find "from parent" confusing since it iterates downwards. Let's just call it `_remove_duplicated_headings`.

```suggestion
    def _remove_duplicated_headings(self, parent: Element) -> bool:
```